userguide/tutorials/overdue.adoc

```diff
@@ -223,95 +251,105 @@ curl -v \
--H 'X-Killbill-CreatedBy: stephane' \
--X POST \
---data-binary '{"systemCommandType":true,"nodeCommandType":"RESTART_PLUGIN","nodeCommandProperties":[{"key":"pluginKey","value":"payment-test"}]}' \
-"http://127.0.0.1:8080/1.0/kb/nodesInfo"
+kpm install_java_plugin 'payment-test' --destination="<path_to_install_plugin>"
 ----
 
+== Testing the System
+
+Now that we have everything ready, we can test the system. We will be simulating the following flow:
+
+image:https://github.com/killbill/killbill-docs/raw/v3/userguide/assets/img/overdue/overdue-test-flow.png[align=center]
 
-At this point, one can create an account and set a payment method to use the `payment test plugin`. The scenario below will do the following:
+So, we will be doing the following:
 
-* Create account
+* Create an account
 * Add default payment matching our test payment plugin
 * Create a subscription
-* Move the clock after the trial and observe first successfull payments
+* Move the clock after the trial and observe first successful payment
 * Configure payment plugin to fail payments
 * Move the clock a month later and observe first failed payment
 * Move clock +1 day and observe first payment retry
 * Move clock +1 day and observe second payment retry
 * Move clock +8 day and observe third payment retry and first overdue state `WARNING`
 * Move clock +4 day and observe second overdue state `BLOCKED`
-* Move clock +7 day and observe second overdue state `CANCELLATION` and verify subscription has been automatically cancelled by the system
+* Move clock +1 day and configure payment plugin to succeed
+* Move clock +1 day (6 days from last payment attempt) and observe fourth payment retry which is successful
 
-Notes: For simplicity, we are using dates (e.g `2016-01-10`) when manipulating the Kill Bill clock instead of fully qualified datetimes (`2016-01-10T01:43:23.000Z`). Passing such a date will end up moving the Kill Bill clock to a given point in time and that point in time may end up before the exact time of the event we are trying to trigger. In such case, retry moving the clock by one day and that should trigger it. An alternative is to specify the exact datetime when moving the clock, but that requires looking into Kill Bill internal tables to understand what is the exact trigger time. This is more rigorous but less convinient.
+Notes: For simplicity, we are using dates (e.g `2021-01-10`) when manipulating the Kill Bill clock instead of fully qualified datetimes (`2021-01-10T01:43:23.000Z`). Passing such a date will end up moving the Kill Bill clock to a given point in time and that point in time may end up before the exact time of the event we are trying to trigger. In such case, retry moving the clock by one day and that should trigger it. An alternative is to specify the exact datetime when moving the clock, but that requires looking into Kill Bill internal tables to understand what is the exact trigger time. This is more rigorous but less convenient.
```

Comment:
FYI I suspect fetching the account and using the reference time attribute from the account object will work in almost all cases (without having to read the database tables).

Reply:
Fixed